pkg/operation/botanist/logging.go

```diff
@@ -70,3 +91,20 @@ func (b *Botanist) DeploySeedLogging(ctx context.Context) error {
 
 	return b.K8sSeedClient.ChartApplier().Apply(ctx, filepath.Join(charts.Path, "seed-bootstrap", "charts", "loki"), b.Shoot.SeedNamespace, fmt.Sprintf("%s-logging", b.Shoot.SeedNamespace), kubernetes.Values(lokiValues))
 }
+
+func (b *Botanist) isShootNodeLoggingActivated() bool {
+	if gardenletfeatures.FeatureGate.Enabled(features.Logging) && b.Shoot != nil &&
+		b.Shoot.Purpose != gardencorev1beta1.ShootPurposeTesting && b.Config != nil &&
```

Comment:
For the first part, to not repeat yourself, what about having a function `IsLoggingEnabled` on the `shootpkg.Shoot` object that does exactly this? You could reuse it here: https://github.com/gardener/gardener/blob/fcf2b21fb7037bca45e17a331225a031718b003e/pkg/operation/botanist/logging.go#L33

Reply:
ok